packages/broker/test/unit/helpers/parser.test.ts

```diff
@@ -0,0 +1,35 @@
+import { parse } from 'qs'
+import { parsePositiveInteger, parseQueryAndBase, parseQueryParameter } from '../../../src/helpers/parser'
+
+describe('parse', () => {
+
+    describe('parseQueryParameter', () => {
+        it('happy path', () => {
+            const query = parse('x=3')
+            const value = parseQueryParameter('x', query, parsePositiveInteger)
+            expect(value).toBe(3)
+        })
+        it('no value', () => {
+            const query = parse('')
+            const value = parseQueryParameter('x', query, parsePositiveInteger)
+            expect(value).toBe(undefined)
+        })
+        it('parse error', () => {
+            const query = parse('x=foo')
+            expect(() => parseQueryParameter('x', query, parsePositiveInteger)).toThrow()
+        })
+    })
+
+    describe('parseQueryAndBase', () => {
+        it('with parameters', () => {
+            const { base, query } = parseQueryAndBase('foobar?lorem=ipsum')
+            expect(base).toBe('foobar')
+            expect(query['lorem']).toBe('ipsum')
+        })
+        it('without parameters', () => {
```

Comment:
Will it handle `parseQueryAndBase('foobar?')` properly?

Reply:
Depends which is expected behavior. If we define that the topic format is `streamId?query`, and query can be empty (like e.g. in http urls), then this return base as `foobar`. We wouldn't support stream ids which contain question marks here. 

I'll add a test case.